oauth2/playground2/src/main/webapp/oauth2.jsp

```diff
@@ -687,10 +687,16 @@
                             <td><input type="submit" name="authorize" value="Get Access Token"></td>
                             <%
                                 if (isOIDCLogoutEnabled) {
+                                    String logout_url = (String)session.getAttribute(OAuth2Constants.OIDC_LOGOUT_ENDPOINT) + "?post_logout_redirect_uri=" + ApplicationConfig.getPostLogoutRedirectUri();
```

Comment:
if we haven't configure PostLogoutRedirectUri what will be the behaviour? Ideally if it is empty we should redirect to default logout page right?

Reply:
At the moment, the `post_logout_redirect_uri` parameter is included in the logout request regardless of whether the `PostLogoutRedirectUri` is provided or not. Therefore, if the `PostLogoutRedirectUri` config is empty, the parameter `post_logout_redirect_uri` will be included in the request without a value, for example `https://<host>:<port>/oidc/logout?post_logout_redirect_uri=&client_id=<client_id_value>`.

If either the client id or token id hint is available, but the the `post_logout_redirect_uri` parameter is empty like `https://<host>:<port>/oidc/logout?post_logout_redirect_uri=&client_id=<client_id_value>`, this will result in an error like the one given below.

```{
  "traceId": "<trace-id>",
  "code": 401,
  "description": "Authorization failure. Authorization information was invalid or missing from your request.",
  "message": "Unauthorized"
}```

However, if neither the client id nor token id hint are available and the `post_logout_redirect_uri` parameter is empty as well, the logout request will be in the format `https://<host>:<port>/oidc/logout?post_logout_redirect_uri=`. Here, the browser will be redirected to the default logout page upon logout. (Note that due to the lack of the client id/token id hint, only Identity Server logout will be possible, while federated IdP will not be possible.)

This will be fixed by conditionally adding the `post_logout_redirect_uri` parameter to the logout request only if the `PostLogoutRedirectUri` is configured, for example like `https://<host>:<port>/oidc/logout?client_id=<client_id_value>` or `https://<host>:<port>/oidc/logout?token_id_hint=<token_id_hint>`, in which case the browser will be redirected to the default logout page.

Reply:
This has been fixed as mentioned in the above comment; if the `PostLogoutRedirectUri` config is empty, the `post_logout_redirect_uri` parameter will not be included and the browser will be redirected to the default logout page.